lib/src/layer/tile_layer.dart

```diff
@@ -317,6 +317,42 @@ class _TileLayerState extends State<TileLayer> with TickerProviderStateMixin {
     _update(null);
     _moveSub = widget.stream.listen((_) => _handleMove());
 
+    _initThrottleUpdate();
+  }
+
+  @override
+  void didUpdateWidget(TileLayer oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    var reloadTiles = false;
+
+    final oldUrl = oldWidget.options.urlTemplate ??
+        oldWidget.options?.wmsOptions?._encodedBaseUrl;
+    final newUrl = options.urlTemplate ?? options?.wmsOptions?._encodedBaseUrl;
+    if (oldUrl != newUrl) {
+      // URL has been changed drop all Tiles and reload them
+      reloadTiles = true;
+    }
+
+    if (oldWidget.options.tileSize != options.tileSize) {
+      // tileSize has been changed drop all Tiles and reload them
+      _tileSize = CustomPoint(options.tileSize, options.tileSize);
+      reloadTiles = true;
+    }
+
+    if (oldWidget.options.updateInterval != options.updateInterval) {
+      // updateInterval has been changed
+      _throttleUpdate?.close();
+      _initThrottleUpdate();
```

Comment:
Shouldn't also set `reloadTiles = true;`?

Reply:
I think not because `updateInterval` is just milliseconds which allows you define how often do you want search for new Tiles (_by default on mobile Leaflet it only loads new tiles when panning ends_).

So basicly if you configure it to `2000ms (2sec)` by default and you change some time later to `200ms (0.2 sec)` then this shouldn't affect already loaded Tiles however it will just run `_update` method (searching) more often.

Reply:
Ohh sure, my bad :p

Good job @maRci002